docs/core/diagnostics/eventsource.md

```diff
@@ -20,6 +20,17 @@ your own custom events.
 > Many technologies that integrate with EventSource use the terms 'Tracing' and 'Traces' instead of 'Logging' and 'Logs'.
 > The meaning is the same here.
 
+> [!WARNING]
+> Potential Deadlocks with EventSource Callbacks
+>
+> When implementing EventSources, be cautious with invoking lock-acquiring APIs within EventSource callbacks. EventSource instances are initialized early in the runtime, before some core features are fully initialized. As a result, the re-entrant behavior of EventSource callbacks may cause deadlocks if the callback attempts to acquire locks already held by the thread that triggered the callback.
+>
+> To mitigate this risk, consider the following precautions:
+>
+> - **Avoid Complex Operations**: Refrain from performing complex operations within the callback that might acquire additional locks.
```

Comment:
```suggestion
> - **Avoid Complex Operations**: Refrain from performing complex operations within the callback that might acquire additional locks. For example during some event callbacks attempting to use File or Console APIs may encounter issues. Instead you can update an in-memory datastructure or add some information about the event to a queue. If more processing is needed it could be done from a separate thread after the callback has already returned.
```

Reply:
Thanks for helping elaborate!